app/controllers/course/users_controller.rb

```diff
@@ -55,6 +55,21 @@ def authorize_edit!
     authorize!(:manage_users, current_course)
   end
 
+  # Updates the course user. This will dispatch an email to the user if he transitions to the
+  # +approved+ state.
+  #
+  # @param [Hash] course_user_params The parameters to set on the given Course User.
+  # @return [bool] True if the course user was updated successfully.
+  def update_course_user(course_user_params)
```

Comment:
It's deliberate. This is pushing the params which the action had down to the logic layer.